pkg/server/v3/server_test.go

```diff
@@ -61,7 +63,52 @@ func (config *mockConfigWatcher) CreateWatch(req *discovery.DiscoveryRequest) (c
 }
 
 func (config *mockConfigWatcher) CreateDeltaWatch(req *discovery.DeltaDiscoveryRequest, st *stream.StreamState) (chan cache.DeltaResponse, func()) {
-	return nil, nil
+	config.counts[req.TypeUrl] = config.counts[req.TypeUrl] + 1
+
+	// Create our out watch channel to return with a buffer of one
+	out := make(chan cache.DeltaResponse, 1)
+
+	if len(config.deltaResponses[req.TypeUrl]) > 0 {
+		res := config.deltaResponses[req.TypeUrl][0]
+		var subscribed []types.Resource
+
+		// Only return back the subscribed resources to our request type
+		r, _ := res.GetDeltaDiscoveryResponse()
+		if len(req.GetResourceNamesSubscribe()) != 0 {
+			for _, resource := range r.Resources {
+				for _, alias := range req.GetResourceNamesSubscribe() {
```

Comment:
Is this right? Seems like this would only work on the requests that add subscriptions, for subsequent requests this would be empty no? Should this use StreamState?

Reply:
I think yes this needs to use the stream state map instead of just looking at the resource subscriptions. As you said that would be empty for subsequent requests